redaxo/src/core/pages/profile.php

```diff
@@ -101,13 +107,20 @@
         $updateuser->setWhere(['id' => $user_id]);
         $updateuser->setValue('password', $userpsw_new_1);
         $updateuser->addGlobalUpdateFields();
+        $updateuser->setDateTimeValue('password_changed', time());
+        $updateuser->setArrayValue('previous_passwords', $passwordPolicy->updatePreviousPasswords($user, $userpsw_new_1));
 
         try {
             $updateuser->update();
             rex_user::clearInstance($user_id);
 
             $success = rex_i18n::msg('user_psw_updated');
 
+            if ($passwordChangeRequired) {
+                $passwordChangeRequired = false;
+                rex::getProperty('login')->setSessionVar('password_change_required', false);
```

Comment:
die storage-daten des logins haben wir durch diesen PR "gut" über die session und die rex-properties verteilt.. 

können wir diese sachen besser in einem neuen objekt/klasse kapseln, sodass dieser ganze code lesbarer wird?

Reply:
Ich habe dafür jetzt Methoden ergänzt.

Reply:
danke, ist schon viel besser in meinen Augen.